### ISSUE TITLE
sgr0 not supported anymore

### ISSUE THREAD

**Issue by edwinvanderham** [jenkinsci/ansicolor-plugin#135]
Using version 0.5.3 of the AnsiColor plugin I could write a script such as:

bold=$(tput -Txterm bold)
norm=$(tput -Txterm sgr0)

echo "${bold}bold text${norm} and normal text"

After upgrading to 0.6.0, the bold text is still bold, but the normal text shows some stange characters followed by a text that is still bold

**Comment by dblock** [comment]
Looks like a regression to me. Appreciate a failing spec if you have time.

**Comment by edwinvanderham** [comment]
What do you mean by spec?

Reproduction scenario is as follows:

- configure any job
- ensure that 'Color ANSI Console Output' is enabled
- add a new build step: 'Execute shell'
```
bold=$(tput -Txterm bold)
norm=$(tput -Txterm sgr0)

echo "${bold}bold text${norm} and normal text"
```
- Build and open console output
- observe that the normal text is in bold

**Comment by dblock** [comment]
An actual test in code in https://github.com/jenkinsci/ansicolor-plugin/tree/master/src/test/java/hudson/plugins/ansicolor.

**Comment by jglick** [comment]
Presumably an effect of #132. Not sure offhand how this scenario would differ from, say, [this](https://github.com/jenkinsci/ansicolor-plugin/blob/39f63c28386f432c4ff17ea4de8f4f2ba21a2089/src/test/java/hudson/plugins/ansicolor/AnsiColorBuildWrapperTest.java#L86) which demonstrates boldfacing. A different escape sequence I suppose?

**Comment by dwnusbaum** [comment]
It seems like a bad interaction with unterminated escape sequences on a single line occuring because the actual escape sequence is printed when the `-x` flag is used. If you inline the variables as follows it works correctly (testing with a Freestyle Job):

    echo "$(tput -Txterm bold) bold text$(tput -Txterm sgr0) and normal text"

Here is a screenshot with the variables inlined: 
<img width="300" alt="screen shot 2018-11-16 at 10 38 29" src="https://user-images.githubusercontent.com/1068968/48631088-d0f14980-e98b-11e8-95c4-7a07dae5106e.png">

Here is a screenshot when they are declared on separate lines:
<img width="306" alt="screen shot 2018-11-16 at 10 39 03" src="https://user-images.githubusercontent.com/1068968/48631141-eb2b2780-e98b-11e8-8ebc-5dd85cc11d75.png">

**Comment by edwinvanderham** [comment]
Yes, that second screenshot is exactly what I am experiencing since the 0.6.0 version

**Comment by dwnusbaum** [comment]
I think I reproduced it in a test in #137, but I'm not entirely sure if my reproduction case is valid. Note that there appears to be another reported regression in #136. Perhaps better to roll back #132 change for now and release 0.6.1 until someone has time to investigate further?

**Comment by dblock** [comment]
Unless this affects everyone in a big way I would only move forward and not rollback anything.

**Comment by jglick** [comment]
If the problem is limited to unterminated escape sequences then this sounds like a duplicate of [JENKINS-34019](https://issues.jenkins-ci.org/browse/JENKINS-34019) perhaps?

**Comment by dwnusbaum** [comment]
> If the problem is limited to unterminated escape sequences then this sounds like a duplicate of JENKINS-34019 perhaps?

Possibly, I now have a fix in-progress in #137 that essentially takes the approach recommended in that ticket, which may fix it, although I have not actually tested it with timestamps.

**Comment by dwnusbaum** [comment]
I just released 0.6.2 with a change that should fix this issue, although you will have to make sure your shell is not echoing the resulting of running `bold=$(tput -Txterm bold)` or it will be interpreted as the beginning of such an escape sequence and subsequent output will be bold.

**Comment by edwinvanderham** [comment]
I can confirm that it works as expected now, thanks for the update
